File: dataset/ZeroShot/GEMINI_gemini-2.5-pro/Task195.py
import heapq

class MedianFinder:
    def __init__(self):
        """
        initialize your data structure here.
        """
        # max-heap for the smaller half, stores negated values
        self.lower_half = []
        # min-heap for the larger half
        self.upper_half = []

    def addNum(self, num: int) -> None:
        # Python's heapq is a min-heap. To simulate a max-heap for the lower half,
        # we store negated numbers.
        
        # Add to the max-heap (lower_half)
        heapq.heappush(self.lower_half, -num)
        
        # Move the largest element from lower_half to upper_half
        heapq.heappush(self.upper_half, -heapq.heappop(self.lower_half))
        
        # Balance the heaps: lower_half should have an equal or one more element
        if len(self.upper_half) > len(self.lower_half):
            heapq.heappush(self.lower_half, -heapq.heappop(self.upper_half))

    def findMedian(self) -> float:
        # If total numbers are odd, the median is the root of the max-heap (lower_half)
        if len(self.lower_half) > len(self.upper_half):
            return float(-self.lower_half[0])
        else:
            # If total numbers are even, median is the average of the two roots
            return (-self.lower_half[0] + self.upper_half[0]) / 2.0

def main():
    # Test Case 1
    print("Test Case 1:")
    medianFinder1 = MedianFinder()
    medianFinder1.addNum(1)
    medianFinder1.addNum(2)
    print(f"Median: {medianFinder1.findMedian()}")  # Output: 1.5
    medianFinder1.addNum(3)
    print(f"Median: {medianFinder1.findMedian()}")  # Output: 2.0
    print()

    # Test Case 2
    print("Test Case 2:")
    medianFinder2 = MedianFinder()
    medianFinder2.addNum(6)
    print(f"Median: {medianFinder2.findMedian()}")  # Output: 6.0
    medianFinder2.addNum(10)
    print(f"Median: {medianFinder2.findMedian()}")  # Output: 8.0
    print()
    
    # Test Case 3
    print("Test Case 3:")
    medianFinder3 = MedianFinder()
    medianFinder3.addNum(1)
    medianFinder3.addNum(2)
    medianFinder3.addNum(3)
    print(f"Median: {medianFinder3.findMedian()}")  # Output: 2.0
    print()
    
    # Test Case 4
    print("Test Case 4:")
    medianFinder4 = MedianFinder()
    medianFinder4.addNum(-1)
    medianFinder4.addNum(-2)
    print(f"Median: {medianFinder4.findMedian()}")  # Output: -1.5
    medianFinder4.addNum(-3)
    print(f"Median: {medianFinder4.findMedian()}")  # Output: -2.0
    medianFinder4.addNum(-4)
    print(f"Median: {medianFinder4.findMedian()}")  # Output: -2.5
    print()
    
    # Test Case 5
    print("Test Case 5:")
    medianFinder5 = MedianFinder()
    medianFinder5.addNum(0)
    print(f"Median: {medianFinder5.findMedian()}")  # Output: 0.0
    print()

if __name__ == "__main__":
    main()

File: dataset/ZeroShot/GEMINI_gemini-2.5-pro/Task195.java
import java.util.Collections;
import java.util.PriorityQueue;

class MedianFinder {
    // max-heap to store the smaller half of the numbers
    private PriorityQueue<Integer> lowerHalf;
    // min-heap to store the larger half of the numbers
    private PriorityQueue<Integer> upperHalf;

    /** initialize your data structure here. */
    public MedianFinder() {
        lowerHalf = new PriorityQueue<>(Collections.reverseOrder());
        upperHalf = new PriorityQueue<>();
    }

    public void addNum(int num) {
        // Add to the max-heap (lower half)
        lowerHalf.offer(num);

        // Move the largest element from lower half to upper half
        upperHalf.offer(lowerHalf.poll());

        // Balance the heaps if upper half becomes larger
        if (upperHalf.size() > lowerHalf.size()) {
            lowerHalf.offer(upperHalf.poll());
        }
    }

    public double findMedian() {
        // If total numbers are odd, median is the top of the lower half (max-heap)
        if (lowerHalf.size() > upperHalf.size()) {
            return lowerHalf.peek();
        } else {
            // If total numbers are even, median is the average of the tops of both heaps
            // Use 2.0 to ensure floating-point division
            return (lowerHalf.peek() + upperHalf.peek()) / 2.0;
        }
    }
}

public class Task195 {
    public static void main(String[] args) {
        // Test Case 1
        System.out.println("Test Case 1:");
        MedianFinder medianFinder1 = new MedianFinder();
        medianFinder1.addNum(1);
        medianFinder1.addNum(2);
        System.out.println("Median: " + medianFinder1.findMedian()); // Output: 1.5
        medianFinder1.addNum(3);
        System.out.println("Median: " + medianFinder1.findMedian()); // Output: 2.0
        System.out.println();

        // Test Case 2
        System.out.println("Test Case 2:");
        MedianFinder medianFinder2 = new MedianFinder();
        medianFinder2.addNum(6);
        System.out.println("Median: " + medianFinder2.findMedian()); // Output: 6.0
        medianFinder2.addNum(10);
        System.out.println("Median: " + medianFinder2.findMedian()); // Output: 8.0
        System.out.println();
        
        // Test Case 3
        System.out.println("Test Case 3:");
        MedianFinder medianFinder3 = new MedianFinder();
        medianFinder3.addNum(1);
        medianFinder3.addNum(2);
        medianFinder3.addNum(3);
        System.out.println("Median: " + medianFinder3.findMedian()); // Output: 2.0
        System.out.println();
        
        // Test Case 4
        System.out.println("Test Case 4:");
        MedianFinder medianFinder4 = new MedianFinder();
        medianFinder4.addNum(-1);
        medianFinder4.addNum(-2);
        System.out.println("Median: " + medianFinder4.findMedian()); // Output: -1.5
        medianFinder4.addNum(-3);
        System.out.println("Median: " + medianFinder4.findMedian()); // Output: -2.0
        medianFinder4.addNum(-4);
        System.out.println("Median: " + medianFinder4.findMedian()); // Output: -2.5
        System.out.println();
        
        // Test Case 5
        System.out.println("Test Case 5:");
        MedianFinder medianFinder5 = new MedianFinder();
        medianFinder5.addNum(0);
        System.out.println("Median: " + medianFinder5.findMedian()); // Output: 0.0
        System.out.println();
    }
}